scripts/check-supabase-migration-order.sh

```
@@ -0,0 +1,148 @@
+
+    existing_added_file="$(awk -F '\t' -v ts="$ts" '$1 == ts { print $2; exit }' "${added_timestamps_file}")"
+    if [[ -n "$existing_added_file" ]]; then
+      echo "❌ Duplicate migration timestamp in this change: ${ts}"
+      echo "  First file: ${existing_added_file}"
+      echo "  Second file: $file"
+      status=1
+    else
+      printf '%s\t%s\n' "$ts" "$file" >> "${added_timestamps_file}"
+    fi
+
+    if [[ "$ts" < "$latest_base_timestamp" ]]; then
```

Shellcheck (0.11.0)

[error] < is for string comparisons. Use -lt instead.

(SC2071)